src/VisualStudio/Core/Def/Implementation/SymbolSearch/SymbolSearchContext.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.VisualStudio.Language.Intellisense.SymbolSearch;
+
+namespace Microsoft.VisualStudio.LanguageServices.SymbolSearch
+{
+    class SymbolSearchContext : FindUsagesContext
+    {
+        private RoslynSymbolSource SymbolSource;
+        private IStreamingSymbolSearchSink SymbolSink;
+
+        public SymbolSearchContext(RoslynSymbolSource symbolSource, IStreamingSymbolSearchSink sink)
+        {
+            this.SymbolSource = symbolSource;
+            this.SymbolSink = sink;
+        }
+
+        public override async Task OnDefinitionFoundAsync(DefinitionItem definition)
+        {
+            return;
+
+            var token = default(CancellationToken);
+            if (definition.SourceSpans.Length == 1)
+            {
+                var result = await RoslynSymbolResult.MakeAsync(this.SymbolSource, definition, definition.SourceSpans[0], token)
+                    .ConfigureAwait(false);
+                this.SymbolSink.Add(result);
+            }
+            else if (definition.SourceSpans.Length == 0)
+            {
+                // that's interesting. let's investigate this!
```

Comment:
this can absolutely happen.  for example a symbol from metadata.

Reply:
yes, we support Go to base to metadata symbols and Go to definition as well

---
In reply to: [338198094](https://github.com/dotnet/roslyn/pull/39462#discussion_r338198094) [](ancestors = 338198094)